sample-app/tests/browser/pagenavigator.js

```diff
@@ -0,0 +1,40 @@
+const { waitTillHTMLRendered } = require('./utils');
+
+/**
+ * Responsible for navigating answers experiences
+ */
+class PageNavigator {
+  /**
+   * @param {import('puppeteer').Page} page A Pupeteer Page
+   * @param {string} siteUrl A url to the index of the site
+   */
+  constructor(page, siteUrl) {
+    this._page = page;
+    this._siteUrl = siteUrl;
+  }
+
+  async gotoPage(path='') {
+    const url = `${this._siteUrl}/${path}`;
+    await this._page.goto(url);
+    await waitTillHTMLRendered(this._page);
+  }
+
+  async click(selector) {
+    await this._page.click(selector);
+    await waitTillHTMLRendered(this._page);
+  }
+
+  async type(input) {
```

Comment:
The `type` and `search` methods seem outside the responsibility of the `PageNavigator` class. These seem more like actions you can perform on the current page, whereas the other two methods can get you from one page to another. Maybe we should consider putting these methods elsewhere or renaming the class.

Reply:
Updated:
 PageNavigator => PageOperator (with all these methods still in one class)
 PageOperator => TestOperator (in change of executing testLocation, now call testInstructions)